keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,1430 @@
+| ------------------------------ | --------------------------------------------------------- | ------------------------------------------------------- |
+| **Cost–reliability trade-off** | Critical workloads stay on premium nodes; batch uses spot | Inference on reliable nodes; training on cheaper pools  |
+| **Workload-aware placement**   | Different claim types target appropriate node tiers       | Pipeline stages match their reliability requirements    |
+| **Graceful preemption**        | `NoExecute` provides controlled eviction timing           | Predictable failover for training and serving workloads |
+| **Resource fairness**          | Prevents monopolization of premium capacity               | Teams share reliable accelerators fairly                |
+| **Elastic scaling**            | Bursts overflow to lower-SLA pools safely                 | HPA scales to spot with clear boundaries                |
+| **Policy transparency**        | Node reliability classes are explicit and auditable       | Platform teams enforce clear reliability tiers          |
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1 — Cluster operator using mixed on-demand and spot nodes
+
+As a cluster operator, I want a default repel from spot (low-SLA) nodes so that only workloads that explicitly tolerate them can land there.
+
+I also want to set numeric SLA thresholds in tolerations (e.g., `Gt 950`) so pods can opt-in to reliable nodes or specific SLA bands without having to hardcode every SLA class in NodeAffinity rules.
+
+**Example Configuration:**
+
+```yaml
+# Spot nodes with 80% SLA get a repelling taint
+apiVersion: v1
+kind: Node
+metadata:
+  name: spot-node-1
+spec:
+  taints:
+  - key: node.kubernetes.io/sla
+    value: "800"
+    effect: NoSchedule
+---
+# Cost-optimized workload explicitly tolerates SLA >= 750
+apiVersion: v1
+kind: Pod
+spec:
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Gt
+    value: "750"
+    effect: NoSchedule
+```
+
+#### Story 2 — AI inference service with strict SLOs
+
+As an AI platform engineer, I want to ensure my latency-critical inference pods only run on nodes with SLA ≥ 95%, and I want them to be evicted if the node's SLA rating drops below that threshold.
+
+Taints and tolerations with numeric comparisons give me this eviction capability, which NodeAffinity cannot provide.
+
+**Example Configuration:**
+
+```yaml
+# High-SLA on-demand node
+apiVersion: v1
+kind: Node
+metadata:
+  name: ondemand-node-1
+spec:
+  taints:
+  - key: node.kubernetes.io/sla
+    value: "950"
+    effect: NoExecute
+---
+# Inference service requires SLA >= 950 with 30s grace period
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inference-service
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node.kubernetes.io/sla
+        operator: Gt
```

Comment:
Should this be Ge?

Reply:
As we agreed on [this](https://github.com/kubernetes/enhancements/pull/5473#discussion_r2269966442) feedback, we removed the `Ge` and `Le` operators.